lib/index.js

```diff
@@ -151,6 +156,39 @@ function lintFile(css, options, filename) {
   return processor.process(css, options.postcss);
 }
 
+/**
+ * Add the encapsulation plugins
+ * to an existing Array of plugins.
+ *
+ * N.B. the plugins must be inserted after
+ * postcss[-easy]-import otherwise encapsulation
+ * is applied only to the entry file selectors.
+ *
+ * @param {Array} plugins Existing Array of plugins or empty Array
+ * @returns {Array} New set of plugins
+ */
+function withEncapsulation(plugins) {
+  var encapsulation = require('./encapsulation');
+  function encapsulationInjectIndex(index, plugin, currentIndex) {
+    if (
+      plugin.postcssPlugin === 'postcss-easy-import' ||
+      plugin.postcssPlugin === 'postcss-import'
+    ) {
+      index = currentIndex + 1;
+    }
+    return index;
+  }
+
+  plugins.splice(
+    plugins.reduce(encapsulationInjectIndex, plugins.length),
+    0,
+    encapsulation.resetGeneric,
+    encapsulation.resetInherited
+  );
+
+  return plugins;
+}
```

Comment:
I think if you use lodash you might still want to test different inputs to the `withEncapsulation` function as you mentioned it behaves incorrectly if `postcss-easy-import` is missing. It'll be easier to pass a few arrays in than using fixtures.